adal/adal-rules.pro

```diff
@@ -0,0 +1,110 @@
+# Add project specific ProGuard rules here.
+# By default, the flags in this file are appended to flags specified
+# in C:\Program Files\Android\android-sdk/tools/proguard/proguard-android.txt
+# You can edit the include path and order by changing the proguardFiles
+# directive in build.gradle.
+#
+# For more details, see
+#   http://developer.android.com/guide/developing/tools/proguard.html
+
+# Add any project specific keep options here:
+
+# If your project uses WebView with JS, uncomment the following
+# and specify the fully qualified class name to the JavaScript interface
+# class:
+#-keepclassmembers class fqcn.of.javascript.interface.for.webview {
+#   public *;
+#}
+
+-dontskipnonpubliclibraryclasses
+
+##---------------Begin: proguard configuration for Common  --------
+-keep,includedescriptorclasses class com.microsoft.identity.common.exception.** { *; }
+-keep,includedescriptorclasses class com.microsoft.identity.common.internal.broker.** { *; }
+-keep,includedescriptorclasses class com.microsoft.identity.common.internal.cache.** { *; }
+-keep,includedescriptorclasses class com.microsoft.identity.logging.internal.dto.** { *; }
+-keep,includedescriptorclasses class com.microsoft.identity.innternal.logging.** { *; }
+-keep,includedescriptorclasses class com.microsoft.identity.common.internal.providers.** { *; }
+-keep,includedescriptorclasses class com.microsort.identity.common.internal.telemtery.** { *; }
+-keep,includedescriptorclasses class com.microsoft.identity.common.internal.ui.** { *; }
+-dontwarn com.microsoft.identity.common.internal.ui.**
+-keep,includedescriptorclasses class com.microsoft.device.display.** { *; }
+-keep,includedescriptorclasses class org.apache.harmony.xnet.provider.jsse.NativeCrypto { *; }
```

Comment:
So we are keeping all these? Why?

Reply:
All of these classes are referred to directly in the ADAL code.

Reply:
That's interesting.....and nothing in the following package is being referred? `com.microsoft.identity.common.adal.internal` (as this package name seems more like it was meant to be for ADAL specific code)......I haven't looked at ADAL code much hence I don't know :D

Reply:
Hm.

Here's what I did (and I thought I put this in the pull-request, but I see that I did not - I'll reconstruct it and add it here because it's important.) - find every import statement and every class reference in ADAL with com.microsoft.identity in it, and find a small set of packages (I won't claim that it's minimal).  Then build the code and find every class that ProGuard complained about and keep it.

Reply:
I recreated what I did and it did identify the .adal packages.  I'm also thinking that the transitive are misplaced - they need to be part of a test app file.  I'm moving them.